fix: recover stuck integrations on server restart

0xSY3 opened the pull request:
**If the server crashes or restarts while a plugin is installing, the integration stays stuck in "InProgress" forever. Users can't do anything with it - can't reinstall, can't uninstall, it's just stuck there.**

**Fix**:
Added a recovery check that runs when the server starts up. It finds any integrations stuck in "InProgress" and marks them as "Failed" so users can see what happened and clean them up.

**Changes**:
  - pkg/workers/recovery.go - new file that checks for stuck integrations on startup and marks them as failed                                                                      
  - main.go - calls the recovery function when server starts                                                                                                                       
  - integration.go (repository) - added started_at timestamp tracking, and allowed uninstall to work on Failed integrations (not just Completed ones)                              
  - integration.go (schema) - added StartedAt field to track when installation began
  
   **Testin**:
  1. Installed System Pulse plugin                                                                                                                                                 
  2. Killed the server while it was installing                                                                                                                                     
  3. Restarted server - integration showed "Failed" with reason "Installation interrupted by server restart"                                                                       
  4. Clicked uninstall - worked and cleaned up resources                                                                                                                           
  5. Verified with kubectl/helm - no leftover namespaces or releases
  
  
  After server restart, integration shows "Failed" status
<img width="449" height="267" alt="1" src="https://github.com/user-attachments/assets/e2d68494-40d6-4132-a50e-fb3dbb218c12" />
Clicking uninstall on failed integration work
<img width="434" height="277" alt="2" src="https://github.com/user-attachments/assets/13f6a9bb-0d90-4231-9cff-5e848091b315" />
 Logs showing uninstall task running and cleanup
<img width="1454" height="608" alt="3" src="https://github.com/user-attachments/assets/07f0772f-c09f-495e-8585-2fc8c5ed7a51" />
No integration namespaces left (cleanup verified)
<img width="745" height="148" alt="Screenshot 2025-12-26 at 4 44 33 AM" src="https://github.com/user-attachments/assets/d4abfcf7-ca70-476f-ae96-3ca0e24caeea" />
No integration helm releases left (cleanup verified)
<img width="1414" height="124" alt="Screenshot 2025-12-26 at 4 46 12 AM" src="https://github.com/user-attachments/assets/d6093919-3428-4ad2-ac5b-bba168dac9d0" />
Shows integration status changed to Terminated
<img width="528" height="295" alt="Screenshot 2025-12-26 at 4 48 51 AM" src="https://github.com/user-attachments/assets/32a66056-5d16-473e-ab70-5110d74aec99" />
UI confirms uninstall completed successfully
<img width="1025" height="63" alt="Screenshot 2025-12-26 at 4 30 26 AM" src="https://github.com/user-attachments/assets/58253bca-8dd0-4157-9b18-069f7c35cd3f" />

0xSY3 committed:
fix: recover stuck integrations on server restart

If server crashes during plugin installation, integrations get stuck in InProgress state. Added startup recovery that marks them as Failed so users can uninstall and retry.

New file pkg/workers/recovery.go handles this. Also added started_at field to track install time and allowed uninstall of failed integrations.

0xSY3 commented on main.go:
```diff
@@ -57,6 +58,9 @@ func main() {
 	}
 
+	//Recovers integrations interrupted by server restart
+	workers.RecoverInterruptedIntegrations(postgresDB)
```
needs to run before server starts accepting requests, otherwise users might see stale InProgress status

0xSY3 commented on pkg/workers/recovery.go:
```diff
@@ -0,0 +1,28 @@
+		Where("status = ?", entities.DeploymentStatusInProgress).
+		Updates(map[string]interface{}{
+			"status": entities.DeploymentStatusFailed,
+			"reason": "Installation interrupted by server restart",
```
its  only used in one place for now

0x6e616d reviewed (commented):
Not only changing the status, I think it's better if we can clean up the resources as well.

0xSY3 committed:
added auto cleanup for interrupted integrations. now when server restarts it cleans up any stuck integrations and removes them from db

0xSY3 committed:
added GetUninstallableIntegration and auto cleanup on install failure as per PR #39 review
also updated all uninstall functions to use GetUninstallableIntegration instead of GetInstalledIntegration
